app/controllers/user_badges_controller.rb

```diff
@@ -109,10 +109,13 @@ def toggle_favorite
       return render json: failed_json, status: 400
     end
 
+    new_is_favorite_value = !user_badge.is_favorite
     UserBadge
       .where(user_id: user_badge.user_id, badge_id: user_badge.badge_id)
-      .update(is_favorite: !user_badge.is_favorite)
+      .update(is_favorite: new_is_favorite_value)
```

Comment:
Sorry not related to your change but while we're here can we handle the possible failure mode here? `update` can fail and we're currently ignoring that case here.

Reply:
Also it looks like running `UserBadge.where().update()` ends up running an update query per `UserBadge` that is returned from the filter. Perhaps we can switch this to use `update_all`?

Reply:
Previously with update(), active record callbacks will run. Even though we might not have callbacks now, someone in the future might come along and add a validation which may cause the update to fail.

Reply:
Oh, I see. But I don't see examples of such checks in our controllers. This kind of validation can be pretty unobvious. For example, in this method we do all validation _before_ calling update which is explicit and perfectly fine in my opinion. 

Frankly I think if someone for some reason decide to do validation in the update hook he should take care of handling this at that moment. If he forgets to handle it, rails will generate and return 500 error to him when he'll be testing his new validation.

Reply:
The problem with not checking for the return value of update() is that the failure becomes silent. A person not working on this method in the controller can add a validation else where for a different purpose. As you’ve said though, the chances of failure is low but the amount of code we have to write to handle failure is so little that it is worth it anyway. We’ve updated the code to use update_all() so the comments here are no longer relevant though.

Reply:
Oh, I get it now, thank you :) `update_all` doesn't instantiate models and doesn't trigger ActiveRecord callbacks.